tests/unit/test_config.py

```diff
@@ -0,0 +1,71 @@
+from bugjira.config import get_config
+
+import pytest
+from pydantic.error_wrappers import ValidationError
+
+
+def test_get_config_good_json(config_defaults):
+    """
+    GIVEN a json file containing a valid definition of a Bugjira config dict
+    WHEN the get_config method is called with its pathname
+    THEN no exception is raised
+    """
+    get_config(config_path=(config_defaults + "/data/config/good_config.json"))
+
+
+def test_get_config_good_dict(good_config_dict):
+    """
+    GIVEN a well-defined dict containing a Bugjira config
+    WHEN the get_config method is called using the dict as the config_dict
+    THEN no exception is raised
+    AND the dict returned is the dict that was passed in
+    """
+    assert get_config(config_dict=good_config_dict) == good_config_dict
+
+
+def test_get_config_missing_values(config_defaults):
+    """
+    GIVEN a json file containing an incomplete definition of a Bugjira config
+    WHEN the get_config method is called with the path to the file
+    THEN a ValidationError is raised
+    """
+    with pytest.raises(ValidationError):
+        get_config(
+            config_path=(config_defaults + "/data/config/missing_config.json")
+        )
+
+
+def test_get_config_no_file(config_defaults):
+    """
+    GIVEN a filepath pointing to a non-existent file
+    WHEN the get_config method is called using the filepath as the config_path
+    THEN a FileNotFoundError is raised
+    """
+    with pytest.raises(FileNotFoundError):
+        get_config(
+            config_path=(config_defaults + "/data/config/does_not_exist.json")
+        )
+
+
+def test_get_config_both_path_and_dict(good_config_file_path):
+    """
+    GIVEN the get_config method, a path to a valid config file, and a dict that
+        varies slightly from the file's config data
+    WHEN the get_config method is called using these non-None values for both
+        the config_path and config_dict parameters
+    THEN the config dict returned by get_config should match the config found
+        in the file referenced by the config_path parameter
+    """
+    # First grab the config_dict from the good config path,
+    # copy it, and change a value in the copy
+    good_config_dict = get_config(good_config_file_path)
+    edited_dict = good_config_dict.copy()
+    edited_dict["bugzilla"] = None
+    # Now call get_config with both the path and the edited dict
+    result = get_config(
+        config_path=good_config_file_path, config_dict=edited_dict
+    )
+    # show that the returned dict from get_config matches the one we get from
+    # the pathname
+    assert result == good_config_dict
+    assert result != edited_dict
```

Comment:
While this test is fine, I wonder if this is really the functionality we want? Maybe it is, but why not allow a dict to be passed in as an override, instead of just ignoring it?

Reply:
I've been thinking about this, and I think I'm coming around to your view here. The config dict seems more 'runtime-y' to me than the path, so I think the dict should prevail. I'll switch this around in the next revision.

Reply:
To clarify, are you intending to do such a change in the next revision of _this_ patch, or in a later patch? Either is ok, I just want to make sure we don't merge something if you plan to change it in this patch.

Reply:
in this patch. working on a refactor as we speak. I'm just that good at multitasking lol.